modin/core/dataframe/algebra/groupby.py

```diff
@@ -655,9 +655,11 @@ def aggregate_on_dict(grp_obj, *args, **kwargs):
                     )
 
             native_res_part = [] if native_agg_res is None else [native_agg_res]
-            result = pandas.concat(
-                [*native_res_part, *custom_results], axis=1, copy=False
-            )
+            parts = [*native_res_part, *custom_results]
+            if parts:
+                result = pandas.concat(parts, axis=1, copy=False)
+            else:
+                result = pandas.DataFrame(columns=result_columns)
```

Comment:
Is it important to keep the column names in case of empty partitions? How did you come to this?

Reply:
I do it to get the expectable result, because result columns are known already. Would it bring any problems or slowdowns?

Reply:
I just want to make sure that Modin's behavior matches that of a pandas. Do we have a test for this new code?